oak_attestation_verification/src/verifier.rs

```diff
@@ -438,7 +425,21 @@ fn verify_root_attestation_signature(
             report.validate().map_err(|msg| anyhow::anyhow!(msg))?;
 
             // Ensure that the attestation report is signed by the VCEK public key.
-            verify_attestation_report_signature(&vcek, report)
+            verify_attestation_report_signature(&vcek, report)?;
+
+            // Check that the root ECA public key for the DICE chain is bound to the attestation
+            // report to ensure that the entire chain is valid.
+            let expected = &hash_sha2_256(&root_layer.eca_public_key[..])[..];
+            let actual = report.data.report_data;
+
+            anyhow::ensure!(
+                // The report data contains 64 bytes by default, but we only use the first 32 bytes
+                // at the moment.
+                expected.len() > actual.len() && expected != &actual[..expected.len()],
+                "The root layer's ECA public key is not bound to the attestation report"
+            );
```

Comment:
this check was originally done at the top level of the verify function. Moving it here imo improves clarity by making the logic of the verify function more seuqential.